File: frontend/lib/core/network/response/bus_route_item_response.dart
class BusRouteItemResponse {

  final String routeKey;

  final String lineName;
  final String operatorName;

  final String originBusStopId;
  final String originName;

  final String destinationBusStopId;
  final String destinationName;

  final String direction;

  BusRouteItemResponse._({
    required this.routeKey,
    required this.lineName,
    required this.operatorName,
    required this.originBusStopId,
    required this.originName,
    required this.destinationBusStopId,
    required this.destinationName,
    required this.direction,
  });

  factory BusRouteItemResponse.fromJson(Map<String, dynamic> json) {
    return BusRouteItemResponse._(
      routeKey: json['routeKey'] as String,
      lineName: json['lineName'] as String,
      operatorName: json['operatorName'] as String,
      originBusStopId: json['originBusStopId'] as String,
      originName: json['originName'] as String,
      destinationBusStopId: json['destinationBusStopId'] as String,
      destinationName: json['destinationName'] as String,
      direction: json['direction'] as String,
    );
  }
}

File: Backend/Backend/Models/Response/Bus/BusRouteItemResponse.cs
namespace Backend.Models
{
    public record BusRouteItemResponse: IResponse
    {
        public required string RouteKey { get; init; }
        public required string LineName { get; init; }
        public required string OperatorName { get; init; }

        public required string OriginBusStopId { get; init; }
        public required string OriginName { get; init; }

        public required string DestinationBusStopId { get; init; }
        public required string DestinationName { get; init; }

        public required string Direction { get; init; }
    }
}
